Grid/qcd/utils/GaugeGroup.h

```diff
@@ -403,31 +426,23 @@ LatticeComplexD Determinant(
 }
 
 template <int N>
-static void ProjectSUn(
-    Lattice<iScalar<iScalar<iMatrix<vComplexD, N> > > > &Umu) {
-  Umu = ProjectOnGroup(Umu);
-  auto det = Determinant(Umu);
-
-  det = conjugate(det);
-
-  for (int i = 0; i < N; i++) {
-    auto element = PeekIndex<ColourIndex>(Umu, N - 1, i);
-    element = element * det;
-    PokeIndex<ColourIndex>(Umu, element, Nc - 1, i);
-  }
+static void ProjectSUn(Lattice<iScalar<iScalar<iMatrix<vComplexD, N> > > > &Umu) {
```

Comment:
Should we only have a single one of these? Roughly like
```c++
template <typename T>
static void ProjectSUn(T &Umu) {
    GaugeGroup<N,GroupName::SU>::ProjectOnSpecialGroup(Umu);
}

Reply:
I tried
`template <typename T, int ncolour>
static void ProjectSUn(T &field) {
    GaugeGroup<ncolour,GroupName::SU>::ProjectOnSpecialGroup(field);
}`

but the compiler can't infer N